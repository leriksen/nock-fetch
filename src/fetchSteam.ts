// if you ask
// tslint:disable-next-line
// http://steamcommunity.com/market/priceoverview/?appid=730&currency=3&market_hash_name=StatTrak%E2%84%A2 M4A1-S | Hyper Beast (Minimal Wear)
//
// you'll get something like
// {"success":true,"lowest_price":"261,35&#8364; ","volume":"11","median_price":"269,52&#8364; "}

import nodeFetch from 'node-fetch';

export interface SteamPriceData {
  lowest_price: string;
  median_price: string;
  success: boolean;
  volume: string;
}

const generateDataUrl = (id: string, name: string) =>
  'https://steamcommunity.com/market/priceoverview/' +
  `?appid=${id}&country=DE&currency=3&market_hash_name=${encodeURIComponent(name.toUpperCase())}`;

export const getData = async (id: string, name: string): Promise<SteamPriceData> => {
  const dataUrl = generateDataUrl(id, name);
  const result = await nodeFetch(dataUrl);
  return await result.json();
};
