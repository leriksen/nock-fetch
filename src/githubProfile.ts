import nodeFetch from 'node-fetch';

const generateDataUrl = (name: string): string => {
  return `https://api.github.com/users/${name}`;
};

export const getData = async (name: string): Promise<string> => {
  const dataUrl = generateDataUrl(name);
  const result = await nodeFetch(dataUrl);
  return await result.json();
};
