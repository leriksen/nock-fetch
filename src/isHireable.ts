// queries github to see if the specifed user has indicated they are available for hire

import { default as fetch } from 'node-fetch';

const generateDataUrl = (name: string): string => {
  return `https://api.github.com/users/${name}`;
};

export const isHireable = async (name: string): Promise<boolean> => {
  const dataUrl = generateDataUrl(name);
  const result = await fetch(dataUrl);
  const json = await result.json();
  return json['hireable'] == true;
};
