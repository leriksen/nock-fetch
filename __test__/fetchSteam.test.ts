import 'jest';

import nock from 'nock';
import { setupRecorder } from 'nock-record';
import { getData } from '../src/fetchSteam';
import nockRecordUncompressed from './__helpers__/nock-record-uncompressed';

const record = setupRecorder();

describe('fetchSteam', () => {
  afterEach(nock.cleanAll);
  afterAll(nock.restore);

  describe('getData', () => {

    it('gets data from Steam', async () => {
      const { completeRecording } = await record(
          'steam-price-data',
          nockRecordUncompressed);
      const result = await getData('578080', 'GAMESCOM INVITATIONAL CRATE');
      completeRecording();
      expect(result).toMatchSnapshot();
    });
  });
});
