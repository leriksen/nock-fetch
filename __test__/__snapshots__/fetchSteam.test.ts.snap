// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchSteam getData gets data from Steam 1`] = `
Object {
  "lowest_price": "0,27€",
  "median_price": "0,27€",
  "success": true,
  "volume": "435",
}
`;
